contracts/src/StabilityPool.sol

```diff
@@ -584,11 +612,11 @@ contract StabilityPool is LiquityBase, IStabilityPool, IStabilityPoolEvents {
 
         Snapshots memory snapshots = depositSnapshots[_depositor];
 
-        uint256 pendingSPYield = activePool.calcPendingSPYield();
+        uint256 pendingSPYield = activePool.calcPendingSPYield() + yieldGainsPending;
```

Comment:
When there are pending yield gains, how are they awarded to depositors?

This function seems to calculate as if all pending gains were immediately awarded pro-rata, including pending yield gains.

But in reality I guess some pending yield gains might not be, right (depending on the SP size). We don't really know when they'll be rewarded. Still, it's fine as an estimate IMO.

Also, I guess and IIRC if there are pending yield gains and a depositor makes a large enough deposit, he receives all those gains, right?